app/assets/javascripts/discourse/app/components/composer-body.js

```diff
@@ -77,6 +77,17 @@ export default Component.extend(KeyEnterEscape, {
     }
   },
 
+  @observes("composeState")
```

Comment:
@CvX Robin used to discourage the use of `observes` but I don't know of a better way to do this. I was wondering if you might have a better idea.

Reply:
I'd move it to `composeStateChanged` (which already is an observer) in composer model. That can function can be refactored when we get to refactoring (moving to a native class and tracked properties)

Reply:
Thank you for the tip @CvX :+1: